taxi/tests/test_views.py

```diff
@@ -0,0 +1,98 @@
+from django.contrib.auth import get_user_model, get_user
+from django.test import TestCase
+from django.urls import reverse
+
+from taxi.models import Manufacturer
+
+
+USER_DATA = {
+    "username": "user1_username",
+    "first_name": "user1_fname",
+    "last_name": "user1_lname",
+    "license_number": "QWE12345",
+    "password": "test12password",
+}
+
+
+class PublicViewsTest(TestCase):
+    def test_redirect_if_not_logged_in(self):
+        paths = [
+            "/",
+            "/cars/",
+            "/drivers/",
+            "/manufacturers/",
+        ]
+        for path in paths:
+            response = self.client.get(path)
+            self.assertEqual(response.status_code, 302)
+            self.assertRedirects(response, f"/accounts/login/?next={path}")
```

Comment:
same here, rewrite without loop
You also should use reverse instead of hardcoding paths and test redirect only using one path (only car-list, for example)